chia/full_node/full_node_api.py

```diff
@@ -312,7 +312,12 @@ async def request_blocks(self, request: full_node_protocol.RequestBlocks) -> Opt
                     return msg
                 block = dataclasses.replace(block, transactions_generator=None)
                 blocks.append(block)
+            msg = make_msg(
+                ProtocolMessageTypes.respond_blocks,
+                full_node_protocol.RespondBlocks(request.start_height, request.end_height, blocks),
+            )
```

Comment:
I don't see why you moved this. there's a path out of the else-block where we don't create an `msg`, we still need to cover that case, no?

Reply:
There are two cases, the `if` and the `else`. In the `if` we have to parse every single block (slower). 
In the `else`, we make a message but directly to bytes, not using streamable.